### PR TITLE
[#572] Simplify senderType

### LINKED ISSUE #572: Simplify senderType
As it's a binary information at first (either the contact sent a message or they didn't), we should reflect this in the data model.

the nuanced data points (which are mostly source specific) should travel through the system via the headers or the metadata

### PR DESCRIPTION
**Pull request by paulodiniz**
Fixes #572 

Screenshot from the chatplugin

Screenshot 2021-04-07 at 14 39 48

Screenshot from the inbox

Screenshot 2021-04-07 at 14 39 54

### CODE REVIEW

File: backend/model/message/src/main/java/co/airy/model/message/MessageRepository.java [hunk 1]
```diff
@@ -23,11 +23,12 @@ public static boolean isNewMessage(Message message) {
...
     // In preparation for https://github.com/airyhq/airy/issues/572
     public static boolean isFromContact(Message message) {
-        return message.getSenderType().equals(SenderType.SOURCE_CONTACT);
+        return message.getIsFromContact();
     }
 
     public static boolean isFromAiry(Message message) {
-        return message.getSenderType().equals(SenderType.APP_USER);
+        return message.getHeaders().get("message.from").equalsIgnoreCase("airy");
+        //return message.getSenderType().equals(SenderType.APP_USER);
```
Comment (paulodiniz) [inline backend/model/message/src/main/java/co/airy/model/message/MessageRepository.java:31]: do we need something like this or can we do 

```java
return !message.getIsFromContact();
```

File: backend/sources/facebook/events-router/src/main/java/co/airy/core/sources/facebook/MessageParser.java [hunk 2]
```diff
@@ -44,42 +43,35 @@ String getSourceConversationId(final JsonNode webhookMessaging) throws NullPoint
...
 
         Optional.ofNullable(postbackNode)
                 .map(node -> node.get("referral"))
                 .ifPresent(referralNode -> headers.put("postback.referral", referralNode.toString()));
 
         return Message.newBuilder()
                 .setContent(payload)
-                .setSenderType(senderType)
                 .setSenderId(senderId)
+                .setIsFromContact(isFromContact)
```
Comment (lucapette) [inline backend/sources/facebook/events-router/src/main/java/co/airy/core/sources/facebook/MessageParser.java:72]: maybe just taste but if the variable is only used here I'd inline the condition

File: lib/typescript/httpclient/model/Content.ts [hunk 3]
```diff
@@ -8,6 +8,10 @@ export interface Content {
...
 export type RenderedContentUnion = Message | Template | Content;
 
-export function isFromContact(message: RenderedContentUnion) {
-  if (message && 'senderType' in message) return message?.senderType === SenderType.sourceContact;
+export function isFromContact(content: RenderedContentUnion) {
+  if (content && 'fromContact' in content) {
+    return content?.fromContact;
+  } else {
+    return false;
+  }
```
Comment (lucapette) [inline lib/typescript/httpclient/model/Content.ts:16]: can do early return here.

Even though I think this function is odd (as it takes a renderedcontentunion but seems message specific anyway. I think @AitorAlgorta and I will address it in #1466)

File: lib/typescript/httpclient/model/Message.ts [hunk 4]
```diff
@@ -16,15 +16,9 @@ export enum MessageState {
...
-export enum SenderType {
-  sourceContact = 'source_contact',
-  sourceUser = 'source_user',
-  appUser = 'app_user',
-}
-
 export interface Message extends Content {
   deliveryState: MessageState;
-  senderType: SenderType;
+  fromContact: boolean;
```
Comment (lucapette) [inline lib/typescript/httpclient/model/Message.ts:21]: 😍

File: backend/sources/twilio/events-router/src/main/java/co/airy/core/sources/twilio/EventsRouter.java [hunk 5]
```diff
@@ -68,7 +67,7 @@ public void startStream() {
                                 messageId,
                                 Message.newBuilder()
                                         .setSenderId(sourceConversationId)
-                                        .setSenderType(SenderType.SOURCE_CONTACT)
+                                        .setIsFromContact(true) //is this true
```
Comment (lucapette) [inline backend/sources/twilio/events-router/src/main/java/co/airy/core/sources/twilio/EventsRouter.java:70]: yes! We should remove the comment though

**Review by lucapette — DISMISSED** [review]
Minor comments

Good stuff 👏

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**